src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -29,27 +29,453 @@ namespace crimson::os::seastore {
 template <typename node_key_t>
 struct FixedKVNode : CachedExtent {
   using FixedKVNodeRef = TCachedExtentRef<FixedKVNode>;
+  struct parent_tracker_t
+    : public boost::intrusive_ref_counter<
+	parent_tracker_t, boost::thread_unsafe_counter> {
+    parent_tracker_t(FixedKVNodeRef parent)
+      : parent(parent) {}
+    parent_tracker_t(FixedKVNode* parent)
+      : parent(parent) {}
+    FixedKVNodeRef parent = nullptr;
+    ~parent_tracker_t() {
+      // this is parent's tracker, reset it
+      if (parent->my_tracker == this) {
+	parent->my_tracker = nullptr;
+      }
+    }
+  };
 
+  using parent_tracker_ref = boost::intrusive_ptr<parent_tracker_t>;
   btree_range_pin_t<node_key_t> pin;
 
-  FixedKVNode(ceph::bufferptr &&ptr) : CachedExtent(std::move(ptr)), pin(this) {}
+  struct copy_source_cmp_t {
+    using is_transparent = node_key_t;
+    bool operator()(const FixedKVNodeRef &l, const FixedKVNodeRef &r) const {
+      assert(l->pin.range.end <= r->pin.range.begin
+	|| r->pin.range.end <= l->pin.range.begin
+	|| (l->pin.range.begin == r->pin.range.begin
+	    && l->pin.range.end == r->pin.range.end));
+      return l->pin.range.begin < r->pin.range.begin;
+    }
+    bool operator()(const node_key_t &l, const FixedKVNodeRef &r) const {
+      return l < r->pin.range.begin;
+    }
+    bool operator()(const FixedKVNodeRef &l, const node_key_t &r) const {
+      return l->pin.range.begin < r;
+    }
+  };
+
+  /*
+   *
+   * Nodes of fixed-kv-btree connect to their child nodes by pointers following
+   * invariants below:
+   *
+   * 1. if nodes are stable:
+   * 	a. parent points at the node's stable parent
+   * 	b. child pointers point at stable children. Child resolution is done
+   * 	   directly via this array.
+   * 	c. copy_sources is empty
+   * 2. if nodes are mutation_pending:
+   * 	a. parent points at stable instance of the same node
```

Comment:
IIUC:
a. parent is empty and needs to be fixed upon commit
b. prior_instance points to its stable version